core/xml.js

```diff
@@ -311,7 +330,7 @@ Blockly.Xml.domToWorkspace = function(xml, workspace) {
       // to be moved to a nested destination in the next operation.
       var block = Blockly.Xml.domToBlock(xmlChild, workspace);
       var blockX = parseInt(xmlChild.getAttribute('x'), 10);
-      var blockY = parseInt(xmlChild.getAttribute('y'), 10);
+      var blockY = parseInt(xmlChild.getAttribute('y'), 10) + bottomY; // add bottomY to be below all blocks
```

Comment:
As Andrew pointed out, some blocks may have a negative 'y' position or the new blocks might start at a very high y position. You'll want your new method to keep a list of all the blocks and the lowest Y value and then add the difference between the lowest y here and the highest y above to every block.

You also probably want to adjust the x values so the new blocks are left aligned with the existing blocks or you could have the same issue of the blocks being very far apart horizontally.